presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestCombineApproxPercentileFunctions.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.sql.parser.ParsingOptions;
+import com.facebook.presto.sql.planner.assertions.PlanMatchPattern;
+import com.facebook.presto.sql.planner.iterative.rule.test.BaseRuleTest;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.DoubleType.DOUBLE;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.aggregation;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.expression;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.project;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.values;
+
+public class TestCombineApproxPercentileFunctions
+        extends BaseRuleTest
+{
+    @Test
+    public void testPercentile()
+    {
+        tester().assertThat(new CombineApproxPercentileFunctions(getMetadata().getFunctionAndTypeManager()))
+                .on(p -> p.aggregation(af -> {
+                    p.variable("col", BIGINT);
+                    af.globalGrouping()
+                            .addAggregation(p.variable("approx_percentile_1"), p.rowExpression("approx_percentile(col, 0.1)", ParsingOptions.DecimalLiteralTreatment.AS_DOUBLE))
+                            .addAggregation(p.variable("approx_percentile_2"), p.rowExpression("approx_percentile(col, 0.2)", ParsingOptions.DecimalLiteralTreatment.AS_DOUBLE))
+                            .source(p.values(p.variable("col")));
+                }))
+                .matchesAnyPattern(
```

Comment:
Can we make this ordering consistent instead of using AnyPattern?

Reply:
The aggregation node stores the output variable reference and its corresponding aggregation expression in a map. Iteration order is not deterministic (and I do see these cases during test). We will need additional sort in the optimization code to make order consistent, which I think maybe not worth the effort just for unit test code.

Reply:
Can you fix the matching logic to do the sorting instead? It should be in AggregationPlanner

Reply:
Added a PR to help this: https://github.com/prestodb/presto/pull/18321

Now you just need to replace:
```
commonApprox.put(dummyAggregation, new HashMap<>());
```

with

```
commonApprox.put(dummyAggregation, new LinkedHashMap<>());
```

Reply:
Thanks for the suggestion. Just updated the diff, and also include the changes in https://github.com/prestodb/presto/pull/18321 to avoid throwing errors for the test.